src/main/java/cow/parser/Parser.java

```diff
@@ -0,0 +1,170 @@
+import cow.commands.FindCommand;
+import cow.commands.HelpCommand;
+import cow.commands.IncorrectCommand;
+import cow.commands.ListCommand;
+import cow.commands.MarkCommand;
+import cow.commands.TodoCommand;
+import cow.commands.UnmarkCommand;
+import cow.exceptions.CowExceptions;
+import cow.message.Messages;
+import cow.tasks.Deadlines;
+
+// solution below inspired by https://github.com/se-edu/addressbook-level2/tree/master
+
+/**
+ * Class to handle the parsing of input commands.
+ **/
+public class Parser {
+    public static final Pattern BASIC_COMMAND_FORMAT = Pattern.compile("(?<commandWord>\\S+)(?<arguments>.*)");
+
+    /**
+     * Parses the string input and returns a command that can be executed
+     * to carry out its task.
+     *
+     * @param userInput full user input from the user.
+     * @return command to be carried out.
+     * @throws CowExceptions used to print any exception that might happen.
+     */
+    public static Command parse(String userInput) throws CowExceptions {
+        Matcher matcher = BASIC_COMMAND_FORMAT.matcher(userInput.trim());
+
+        if (!matcher.matches()) {
+            return new IncorrectCommand(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT,
+                    HelpCommand.MESSAGE_USAGE));
+        }
+
+        final String commandWord = matcher.group("commandWord");
+        final String arguments = matcher.group("arguments");
+
+        switch (commandWord) {
+        case ListCommand.COMMAND_WORD:
+            return new ListCommand();
+        case ByeCommand.COMMAND_WORD:
+            return new ByeCommand();
+        case TodoCommand.COMMAND_WORD:
+            return prepareTodo(arguments);
+        case DeadlineCommand.COMMAND_WORD:
+            return prepareDeadline(arguments);
+        case EventCommand.COMMAND_WORD:
+            return prepareEvent(arguments);
+        case MarkCommand.COMMAND_WORD:
+            return new MarkCommand(getMarkUnmarkInt(arguments));
+        case UnmarkCommand.COMMAND_WORD:
+            return new UnmarkCommand(getMarkUnmarkInt(arguments));
+        case DeleteCommand.COMMAND_WORD:
+            return new DeleteCommand(getMarkUnmarkInt(arguments));
+        case DueCommand.COMMAND_WORD:
+            return prepDue(arguments);
+        case FindCommand.COMMAND_WORD:
+            if (arguments == null || arguments.isEmpty()) {
+                return new IncorrectCommand(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT,
+                        FindCommand.MESSAGE_USAGE));
+            }
+            return new FindCommand(arguments);
+        default:
+            return new HelpCommand();
+        }
+    }
+
+    /**
+     * Parses the index used for marking, unmarking and delete commands.
+     *
+     * @param args provided by command.
+     * @return the parse int.
+     * @throws CowExceptions exceptions due to invalid index.
+     */
+    private static int getMarkUnmarkInt(String args) throws CowExceptions {
```

Comment:
I'm not really sure if this method names clearly describes what it does. I think something like "getIndexFromCommand" might work better